syncer/state_syncer.go

```diff
@@ -132,32 +139,77 @@ func (s *Syncer) needCert(ctx context.Context, lid types.LayerID) (bool, error)
 	return errors.Is(err, sql.ErrNotFound), nil
 }
 
-func (s *Syncer) fetchOpinions(ctx context.Context, lid types.LayerID) ([]*fetch.LayerOpinion, error) {
+func (s *Syncer) fetchOpinions2(ctx context.Context, lid types.LayerID) ([]*peerOpinion, []*types.Certificate, error) {
+	s.logger.WithContext(ctx).With().Debug("polling layer opinions", lid)
+	needCert, err := s.needCert(ctx, lid)
+	if err != nil {
+		return nil, nil, err
+	}
+	opinions, certs, err := s.dataFetcher.PollLayerOpinions2(ctx, lid, needCert)
+	if err != nil {
+		s.logger.WithContext(ctx).With().Warning("failed to fetch opinions", lid, log.Err(err))
+		return nil, nil, fmt.Errorf("PollLayerOpinions: %w", err)
+	}
+	var result []*peerOpinion
+	for _, opn := range opinions {
+		result = append(result, &peerOpinion{
+			prevAggHash: opn.PrevAggHash,
+			peer:        opn.Peer(),
+		})
+		if opn.CertBlock != types.EmptyBlockID {
+			s.dataFetcher.RegisterPeerHashes(opn.Peer(), []types.Hash32{opn.CertBlock.AsHash32()})
+		}
+	}
+	opinionLayer.Set(float64(lid))
+	return result, certs, nil
+}
+
+func (s *Syncer) fetchOpinions(ctx context.Context, lid types.LayerID) ([]*peerOpinion, []*types.Certificate, error) {
+	if s.ticker.CurrentLayer() >= types.LayerID(s.cfg.UpdateLayer) {
```

Comment:
used PeerStore().GetProtocols() to make upgrade path automatic.
peer will first fetch from v2 peers, and then fetch from peers that don't support v2
as this is not on critical path, i did this sequentially instead of concurrently for simpler code.